frontend/src/components/Common/FirestoreDataProvider.tsx

```diff
@@ -102,16 +109,32 @@ export default function FirestoreDataProvider({ children }: Props): JSX.Element
         // Do not run firestore listeners in test environment.
       };
     }
-    MembersAPI.hasIDOLAccess(userEmail).then((hasIDOLAccess) => setHasIDOLAccess(hasIDOLAccess));
-    const unsubscriberOfAdminEmails = onSnapshot(adminsCollection, (snapshot) => {
-      setAdminEmails(snapshot.docs.map((it) => it.id));
-    });
-    const unsubscriberOfMembers = onSnapshot(membersCollection, (snapshot) => {
-      setMembers(snapshot.docs.map((it) => it.data()));
-    });
-    const unsubscriberOfApprovedMembers = onSnapshot(approvedMembersCollection, (snapshot) => {
-      setApprovedMembers(snapshot.docs.map((it) => it.data()));
+    MembersAPI.hasIDOLAccess(userEmail, 'applicants-included').then((hasIDOLAccess) => {
+      setHasIDOLAccess(hasIDOLAccess);
     });
+    const unsubscriberOfAdminEmails = onSnapshot(
+      adminsCollection,
+      (snapshot) => {
+        setAdminEmails(snapshot.docs.map((it) => it.id));
+      },
+      (error) => {
```

Comment:
Awesome job including the error handling! What are your thoughts about adding retry logic for the error handling? I was just wondering since I don't know how often Firebase run sinto issues with real-time data? Wanted to know your thoughts on this! 
```suggestion
      let retries = 0 
      (error) => {
            console.error("Error fetching admin emails:", error);
            if (retries < 2) {
              retries++;
              setTimeout(fetchAdminEmails, 2000); // Retry after 2 seconds or however long we want to wait haha :)
            } else {
              alert("Failed to fetch the Admin Emails after Retry");
            }
```

Reply:
Awesome that sounds great! Feel free to ignore this comment!